test_harness/single_node/install.sh

```diff
@@ -0,0 +1,159 @@
+  esac
+done
+
+
+run_phase() {
+
+ local PHASE=$1
+ local with_opts=" $2 "
+
+ case "$PHASE" in
+
+ 0)
+
+    if [[ $with_opts = *\ initialize\ * ]]; then
+        apt-get -y update
+        apt-get  install -y apt-transport-https wget lsb-release sudo jq
+    fi
+
+    if [[ $with_opts = *\ sudo-without-pw\ * ]]; then
+      if [ $(id -u) = 0 -a "${USER:-root}" = root ] ; then
+        echo >&2 "root authorization for 'sudo' is automatic - no /etc/sudoers modification needed"
+      else
+        if [ -f "/etc/sudoers" ]; then
+            # add a line with our USER name to /etc/sudoers if not already there
+            sudo su -c "sed -n '/^\s*[^#]/p' /etc/sudoers | grep '^$USER\s*ALL=(ALL)\s*NOPASSWD:\s*ALL\s*$' >/dev/null" || \
+            sudo su -c "echo '$USER ALL=(ALL) NOPASSWD: ALL' >>/etc/sudoers"
+        else
+            echo >&2 "WARNING - Could not modify sudoers files"
+        fi
+      fi # not root
+    fi # with-opts
+
+    #------ (needed for both package install and build from source)
+
+    if [[ $with_opts = *\ install-essential-packages\ * ]]; then
+
+        if ! dpkg -l tzdata >/dev/null 2>&1 ; then
+          sudo su - root -c \
+           "env DEBIAN_FRONTEND=noninteractive bash -c 'apt-get install -y tzdata'"
+        fi
+        sudo apt-get update
+        sudo apt-get install -y software-properties-common postgresql
+        sudo apt-get update && \
+        sudo apt-get install -y libfuse2 unixodbc rsyslog
+    fi
+
+
+    if [[ $with_opts = *\ add-package-repo\ * ]]; then
+            add_package_repo -f
+    fi
+
+
+    if [[ $with_opts = *\ create-db\ * ]]; then
+    sudo su - postgres -c "
+        { dropdb --if-exists ICAT
+          dropuser --if-exists irods ; } >/dev/null 2>&1"
+    sudo su - postgres -c "psql <<\\
+________
+        CREATE DATABASE \"ICAT\";
+        CREATE USER irods WITH PASSWORD 'testpassword';
+        GRANT ALL PRIVILEGES ON DATABASE \"ICAT\" to irods;
+________"
+    echo >&2 "-- status of create-db =  $? -- "
+    fi
+    ;;
+
+ 4)
+   IRODS_TO_INSTALL=$(irods_package_vsn)
+   sudo apt install -y irods-{dev,runtime}${IRODS_TO_INSTALL:+"=$IRODS_TO_INSTALL"}
+   if [[ $with_opts != *\ basic\ * ]]; then
+     sudo apt install -y irods-{icommands,server,database-plugin-postgres}${IRODS_TO_INSTALL:+"=$IRODS_TO_INSTALL"}
+   fi
+ ;;
+
+ 5)
+ if [ ! $(irods_vsn) '<' "4.3" ]; then
+    PYTHON=python3
+ else
+    PYTHON=python2
+ fi
+ sudo $PYTHON /var/lib/irods/scripts/setup_irods.py < /var/lib/irods/packaging/localhost_setup_postgres.input
+ ;;
+
+ *) echo >&2 "unrecognized phase: '$PHASE'." ; QUIT=1 ;;
+ esac
+ return $?
+}
+
+#-------------------------- main
+
+QUIT=0
+while [ $# -gt 0 ] ; do
+  ARG=$1 ; shift
+  NOP="" ; run_phase $ARG " $withopts "; sts=$?
+  [ $QUIT != 0 ] && break
+  [ -n "$NOP" ] && continue
+  echo -n "== $ARG == "
+  if [ $sts -eq 0 ]; then
+    echo Y >&2
+  else
+    [ $quit_on_phase_err ] && { echo >&2 "N - quitting"; exit 1; }
+    echo N >&2
+  fi
```

Comment:
Indentation is different from lines preceding these.

Reply:
please elaborate.  It looks ok to me....

Reply:
Interesting! Will investigate.

Reply:
Do we need a GitHub action to reformat the Python bits too?

Reply:
We have https://github.com/irods/python-irodsclient/issues/726 for that.